addons/sale_project/models/project_project.py

```diff
@@ -138,7 +152,11 @@ def action_customer_preview(self):
 
     @api.onchange('reinvoiced_sale_order_id')
     def _onchange_reinvoiced_sale_order_id(self):
-        if not self.sale_line_id and self.reinvoiced_sale_order_id.order_line:
+        if (
+            not self.sale_line_id
+            and self.reinvoiced_sale_order_id.order_line
+            and self.reinvoiced_sale_order_id.order_line[0].is_service
+        ):
             self.sale_line_id = self.reinvoiced_sale_order_id.order_line[0]
```

Comment:
We could maybe filtered the SOLs inside `reinvoiced_sale_order_id` to be able to get the first service found, no? 🙂 

```suggestion
        if (
            not self.sale_line_id
            and (service_sols := self.reinvoiced_sale_order_id.order_line.filtered('is_service'))
        ):
            self.sale_line_id = service_sols[0]
```

what do you think? 🙂

Reply:
Yes, that makes sense! I'll go ahead with this change. Thanks for the suggestion! 😊